solr/core/src/java/org/apache/solr/search/TermsQParserPlugin.java

```diff
@@ -79,20 +88,34 @@ Query makeFilter(String fname, BytesRef[] byteRefs) {
     },
     automaton {
       @Override
-      Query makeFilter(String fname, BytesRef[] byteRefs) {
+      Query makeFilter(String fname, BytesRef[] byteRefs, SolrParams localParams) {
         ArrayUtil.timSort(byteRefs); // same sort algo as TermInSetQuery's choice
         Automaton union = Automata.makeStringUnion(Arrays.asList(byteRefs)); // input must be sorted
         return new AutomatonQuery(new Term(fname), union);//constant scores
       }
     },
     docValuesTermsFilter {//on 4x this is FieldCacheTermsFilter but we use the 5x name any way
       @Override
-      Query makeFilter(String fname, BytesRef[] byteRefs) {
-        return new DocValuesTermsQuery(fname, byteRefs);//constant scores
+      Query makeFilter(String fname, BytesRef[] byteRefs, SolrParams localParams) {
+        final String type = localParams.get(SUBMETHOD);
```

Comment:
Instead of yet another local-param, maybe "method" could be docValuesTermsFilterTopLevel and docValuesTermsFilterTopLevel.  The existing docValuesTermsFilter could dispatch to either of those.